### ISSUE TITLE
[3.0.1] destDir not respected

### ISSUE THREAD

**Issue by jrowlingson** [broccolijs/broccoli-funnel#122]
It appears that as of version 3.0.1 the `destDir` is no longer honored. In the reproduction below, the `app.css` file is correctly written to `dist/assets/styles/` while the 3.0.1 example is writing the file to `dist/`.

### Brocfile.js
```
const funnel = require('broccoli-funnel')
module.exports = funnel('app/styles', {
  files: ['app.css'],
  destDir: '/assets/styles',
})
```
### 3.0.0

```
$ tree app
app
├── index.html
└── styles
    └── app.css

1 directory, 3 files

$ rm -rf dist && broccoli build; tree dist
dist
└── assets
    └── styles
        └── app.css

2 directories, 1 file
```
### 3.0.1
```
$ tree app
app
├── index.html
└── styles
    └── app.css

1 directory, 3 files

$ rm -rf dist && broccoli build; tree dist
dist
├── app.css
└── assets
    └── styles

2 directories, 1 file
```

**Comment by rwjblue** [comment]
@SparshithNR - Can you look into this? This seems related to the work done in #120.

**Comment by bendemboski** [comment]
I'm seeing this as well -- staying pinned to 3.0.0 for now as it's pretty fatal to my build!

**Comment by SparshithNR** [comment]
That's my bad. I have made a code change and added a test case for covering this case.

**Comment by eliel-jr** [comment]
I'd same problem after update broccoli-funnel from 2.0.2 to 3.0.2.

**Comment by sebastianhelbig** [comment]
I can confirm same problem with 3.0.2

**Comment by drewlee** [comment]
Ditto, still experiencing this issue in v3.0.2.

**Comment by mmun** [comment]
The fix was released in v3.0.3 and I've confirmed it works in my app.

Thanks @SparshithNR and @rwjblue!
